src/api/cvc4cpp.h

```diff
@@ -0,0 +1,4556 @@
+   * Parameters: 1
+   *   -[1]: the Boolean Term to negate
+   * Create with:
+   *   mkTerm(Kind kind, Term child)
+   */
+  NOT,
+  /* Logical and.
+   * Parameters: n > 1
+   *   -[1]..[n]: the Boolean Term of the conjunction
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  AND,
+  /**
+   * Logical implication.
+   * Parameters: 2
+   *   -[1]..[2]: Boolean Terms, [1] implies [2]
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  IMPLIES,
+  /* Logical or.
+   * Parameters: n > 1
+   *   -[1]..[n]: the Boolean Term of the disjunction
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  OR,
+  /* Logical exclusive or.
+   * Parameters: 2
+   *   -[1]..[2]: Boolean Terms, [1] xor [2]
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  XOR,
+  /* If-then-else.
+   * Parameters: 3
+   *   -[1] is a Boolean condition Term
+   *   -[2] the 'then' Term
+   *   -[3] the 'else' Term
+   * 'then' and 'else' term must have same base sort.
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  ITE,
+
+  /* UF -------------------------------------------------------------------- */
+
+  /* Application of an uninterpreted function.
+   * Parameters: n > 1
+   *   -[1]: the function Term
+   *   -[2]..[n]: function argument instantiation Terms
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  APPLY_UF,
+  /**
+   * Boolean term variable
+   * ???
+   */
+  BOOLEAN_TERM_VARIABLE,
+  /**
+   * Cardinality constraint on sort S.
+   * Parameters: 2
+   *   -[1]: a Term of Sort S
+   *   -[2]: a positive integer constant that bounds the cardinality of sort S
+   * Create with:
+   *   ???
+   */
+  CARDINALITY_CONSTRAINT,
+  /**
+   * Combined cardinality constraint.
+   * Parameters: 1
+   *   -[1]: a positive integer constant that bounds the sum of the
+   *         cardinalities of all sorts in the signature
+   * Create with:
+   *   ???
+   */
+  COMBINED_CARDINALITY_CONSTRAINT,
+  /**
+   * Partial uninterpreted function application.
+   * Parameters: n  ??? > 1 ???
+   *   -[1]..[n]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  PARTIAL_APPLY_UF,
```

Comment:
I think these are all internal only: CARDINALITY_CONSTRAINT, COMBINED_CARDINALITY_CONSTRAINT, PARTIAL_APPLY_UF, @ajreynol can you confirm?

Reply:
There is a public interface for CARDINALITY_CONSTRAINT in the smt2 parser actually and is supported by cardinality extension of the UF solver.  I had some requests for this.

COMBINED_CARDINALITY_CONSTRAINT and PARTIAL_APPLY_UF are internal.